clients/src/main/java/org/apache/kafka/common/network/ServerConnectionId.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.network;
+
+import java.net.Socket;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * ServerConnectionId is used to uniquely identify a connection on server for the client. The
+ *  connection id is in the format of "localHost:localPort-remoteHost:remotePort-processorId-index".
+ *  The processorId is the id of the processor that will handle this connection and the index is
+ *  used to ensure uniqueness.
+ */
+public class ServerConnectionId {
+
+    // The regex for parsing the URI string. The URI string should be in the format of "host:port",
+    //  where host can be an IPv4 address or an IPv6 address.
+    //  Note: The IPv6 address should not be enclosed in square brackets.
+    private static final Pattern URI_PARSE_EXP = Pattern.compile("([0-9a-zA-Z\\-%._:]*):([0-9]+)");
```

Comment:
Since this is not really a URI, should we rename URI_PARSE_EXP to sth like HOST_PORT_PARSE_EXP?

Reply:
Done.